### PR TITLE
TIP-1111: use ES aliases instead of indexes

### PR DESCRIPTION
**Pull request by ahocquard**
**Description (for Contributor and Core Developer)**

It uses aliases instead of indexes for all the indexes, in order to be consistent everywhere.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Y
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | Todo with Benoit
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/Client.php [hunk 1]
```diff
@@ -252,9 +254,10 @@ public function createIndex()
     {
         $configuration = $this->configurationLoader->load();
         $body = $configuration->buildAggregated();
+        $body['aliases'] = [$this->indexName => (object) []];
 
         $params = [
-            'index' => $this->indexName,
+            'index' => $this->indexName . '_' . str_replace('.', '_', CommunityVersion::VERSION) . '_' . Uuid::uuid4(),
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/ElasticsearchBundle/Client.php]: You could use an sprintf here to improve readability

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/IndexConfiguration/UpdateIndexMapping.php [hunk 2]
```diff
@@ -27,7 +28,7 @@ final class UpdateIndexMapping
     public function updateIndexMapping(Client $client, string $indexNameOrAlias, Loader $indexConfiguration): void
     {
         // We don't care about the index name anymore as we use alias
-        $newIndexName = $indexNameOrAlias . '_' . str_replace('.', '_', CommunityVersion::VERSION) . '_' . time();
+        $newIndexName = $indexNameOrAlias . '_' . str_replace('.', '_', CommunityVersion::VERSION) . '_' . Uuid::uuid4();
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/ElasticsearchBundle/IndexConfiguration/UpdateIndexMapping.php]: sprintf also here

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php [hunk 3]
```diff
@@ -58,21 +58,20 @@ public function test_it_is_idempotent()
...
         $this->createProduct('product1');
         Assert::assertEquals(1, $pqb->create()->execute()->count());
-        Assert::assertContains($this->getParameter('product_index_name'), $indices);
+        Assert::assertContains($this->getParameter('product_index_name'), $aliases);
 
 
         $updateIndexMapping = new UpdateIndexMapping();
-        // from index to alias
+        // from alias to alias
         $updateIndexMapping->updateIndexMapping($client, $akeneoProductClient->getIndexName(), $akeneoProductClient->getConfigurationLoader());
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php]: Is it normal that those two lines are the same ?

Comment (anaelChardan) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php]: Yup it is to test the migration on multiple sources (one being an index and the second one being an alias)

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/IndexConfiguration/UpdateIndexMapping.php (hunk 2 above)
Comment (anaelChardan) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/IndexConfiguration/UpdateIndexMapping.php]: Why UUID?

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/Client.php (hunk 1 above)
Comment (ahocquard) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/Client.php]: I don't think that '%s_%s_%s_%s' is much more readable

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/IndexConfiguration/UpdateIndexMapping.php (hunk 2 above)
Comment (ahocquard) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/IndexConfiguration/UpdateIndexMapping.php]: avoid conflict when creating at the same time the index for free

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php (hunk 3 above)
Comment (ahocquard) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php]: test idempotency

**Pull request merged**